openlibrary/plugins/openlibrary/js/autocomplete.js

```diff
@@ -16,59 +16,118 @@ export default function($) {
      * @param{HTMLInputElement} _this - input element that will become autocompleting.
      * @param{OpenLibraryAutocompleteOptions} ol_ac_opts
      * @param{Object} ac_opts - options passed to $.autocomplete; see that.
+     * @param {Function} ac_opts.formatItem - optional item formatter. Returns a string of HTML for rendering as an item.
      */
     function setup_autocomplete(_this, ol_ac_opts, ac_opts) {
         var default_ac_opts = {
             autoFill: true,
-            mustMatch: true,
-            formatMatch: function(item) { return item.name; },
-            parse: function(text) {
-                // in v2, text IS the JSON
-                var rows = typeof text === 'string' ? JSON.parse(text) : text;
-                var parsed = [];
-                var i, row, query;
-                for (i=0; i < rows.length; i++) {
-                    row = rows[i];
-                    parsed.push({
-                        data: row,
-                        value: row.name,
-                        result: row.name
-                    });
-                }
-
-                // XXX: this won't work when _this is multiple values (like $("input"))
-                query = $(_this).val();
-                if (ol_ac_opts.addnew == true || (ol_ac_opts.addnew && ol_ac_opts.addnew(query))) {
-                    parsed = parsed.slice(0, ac_opts.max - 1);
-                    const name = ol_ac_opts.new_name || query;
-                    parsed.push({
-                        data: {name, key: '__new__'},
-                        value: name,
-                        result: name
+            formatItem: function (item) {
+                return item.name;
+            },
```

Comment:
Done. I highly recommend you create a github issue adding an eslint rule if that's the preferred style for the repo. (https://eslint.org/docs/rules/prefer-arrow-callback for example).

Code style shouldn't really come up in code reviews. The onus should be on the maintainers to define these. These sort of things can and should be automated. It will save us a lot of time with new contributors less familiar with our codebase by allowing us to focus on functionality in code reviews.